packages/back-end/src/models/ExperimentModel.ts

```diff
@@ -499,15 +520,31 @@ export async function getRecentExperimentsUsingMetric(
 }
 
 export async function deleteExperimentSegment(
-  organization: string,
+  organization: OrganizationInterface,
   segment: string
 ): Promise<void> {
-  await ExperimentModel.updateOne(
-    { organization, segment },
+  const exps = await getExperimentsUsingSegment(segment, organization.id);
+
+  if (!exps.length) return;
+
+  await ExperimentModel.updateMany(
+    { organization: organization.id, segment },
     {
       $set: { segment: "" },
     }
   );
+
+  exps.forEach((previous) => {
+    const current = cloneDeep(previous);
+    current.segment = "";
+
+    // TODO Confirm we want to fire webhooks here?
+    _onExperimentUpdate({
+      organization,
+      oldExperiment: previous,
+      newExperiment: current,
```

Comment:
Q: Does deleting a segment for experiments require we trigger webhoos/refresh CDN?